importer/balanced/builder.go

```diff
@@ -1,24 +1,40 @@
-// Package balanced provides methods to build balanced DAGs.
-// In a balanced DAG, nodes are added to a single root
-// until the maximum number of links is reached (with leaves
-// being at depth 0). Then, a new root is created, and points to the
-// old root, and incorporates a new child, which proceeds to be
-// filled up (link) to more leaves. In all cases, the Data (chunks)
-// is stored only at the leaves, with the rest of nodes only
-// storing links to their children.
+// Package balanced provides methods to build balanced DAGs. In a
+// balanced DAG, nodes are added to a single root until the maximum
+// number of links is reached (with leaves being at depth 0). Then, a new
+// root is created to point to the old root (incorporating it as its
+// first child), which proceeds to be filled up (link) to more nodes. In
+// all cases, the data (chunk) is stored only at the leaves, with the
+// rest of (internal) nodes only storing links to their children and
+// sizes of the data stored under them.
 //
-// In a balanced DAG, nodes fill their link capacity before
-// creating new ones, thus depth only increases when the
-// current tree is completely full.
+// Internal nodes are always represented by UnixFS nodes (of type `File`)
+// encoded inside DAG nodes, in contrast, leaf nodes with data have
+// multiple possible representations: UnixFS nodes as above, raw nodes
+// with just the file data (no format) and Filestore nodes (supported by
+// raw nodes). In the case the entire file fits into just one node it
+// will be formatted as a (single) leaf node (without parent) with the
+// possible representations already mentioned, this is the only scenario
+// where the root can be of a type different that the UnixFS node.
```

Comment:
> I’m not sure what “Filestore nodes (supported by raw nodes)” means.

I left a vague description on purpose because I don't know exactly how Filestore operates, just that it's stored on raw nodes, *support* in communication protocols terminology is which type of lower layer packet carries your data, but I can just say that it's a format stored on a raw node.



> In “possible representations already mentioned, this is the only scenario”, replace the comma with a period or put the “this is the only scenario…” part in parentheses. That helps keep each sentence to one clear idea and makes them much more readable.
> 
> It also might be helpful to say “(see the go-ipfs/unixfs package for details of UnixFS.)” at the end. Or something like that to point to more details on UnixFS.

:+1: